go/fixchain/hash.go

```diff
@@ -0,0 +1,60 @@
+package fixchain
+
+import (
+	"crypto/sha256"
+	"sort"
+
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+const hashSize = sha256.Size
+
+var newHash = sha256.New
+
+func hash(c *x509.Certificate) (hash [hashSize]byte) {
+	copy(hash[:], newHash().Sum(c.Raw))
+	return
+}
+
+func hashChain(ch []*x509.Certificate) (hash [hashSize]byte) {
+	h := newHash()
+	for _, c := range ch {
+		h.Write(newHash().Sum(c.Raw))
+	}
+	copy(hash[:], h.Sum(nil))
+	return
+}
+
+func hashBag(chain []*x509.Certificate) [hashSize]byte {
```

Comment:
It's not a precondition, no.  The point of hashBag is just to hash a given set of certificates, whether it contains duplictates or not.  It was used for caching in the fixer, but I've since removed that, and am currently undecided about where and how it should be put back, if at all.  If we were to put it back it'd probably be at the FixAndLog level.